src/main/java/org/ow2/proactive/catalog/rest/controller/BucketController.java

```diff
@@ -109,12 +109,27 @@ public BucketMetadata create(
             throws NotAuthenticatedException, AccessDeniedException {
         String initiator = ANONYMOUS;
         if (sessionIdRequired) {
-            restApiAccessService.checkAccessBySessionIdForOwnerOrGroupAndThrowIfDeclined(sessionId, ownerName);
+            AuthenticatedUser authenticatedUser = restApiAccessService.getUserFromSessionId(sessionId);
+            String tenant = authenticatedUser.getTenant();
+            if (!Strings.isNullOrEmpty(tenant)) {
+                restApiAccessService.checkAccessBySessionIdForOwnerOrGroupAndThrowIfDeclined(sessionId, ownerName);
+            } else {
+                restApiAccessService.checkAccessBySessionIdForOwnerOrGroupOrTenantAndThrowIfDeclined(sessionId,
+                                                                                                     ownerName,
+                                                                                                     restApiAccessService.getUserFromSessionId(sessionId)
+                                                                                                                         .getTenant());
+            }
             initiator = restApiAccessService.getUserFromSessionId(sessionId).getName();
         }
         try {
-            BucketMetadata bucketMetadata = bucketService.createBucket(bucketName, ownerName);
-            log.info(ACTION + initiator + " created new bucket " + bucketName + " with owner " + ownerName);
+            String tenant = "";
+            if (sessionIdRequired) {
+                AuthenticatedUser authenticatedUser = restApiAccessService.getUserFromSessionId(sessionId);
+                tenant = authenticatedUser.getTenant();
```

Comment:
tenant = authenticatedUser.getTenant() may return null, while you initialize tenant to empty string ...